### PR TITLE
fix(documentation): create testing documentation (#371)

### LINKED ISSUE #371: Create Testing Documentation - Development
In this task we want to fill out the template for the TESTING.md from the documentation skeleton.

----
**ACs**
- [ ] The [TESTING.md](https://github.com/deven-org/B01LER-Kitchen/blob/develop/doc/TESTING.md) template has been filled out

----
**Background informatione**
Filled out templates:
- [Deven](https://github.com/deven-org/documentation-skeleton/blob/main/doc/TESTING.md)
- [Telemetry](https://github.com/deven-org/telemetry-functions/blob/main/doc/TESTING.md)

### PR DESCRIPTION
**Pull request by angsherpa456**
closes #371

### CODE REVIEW

File: TESTING.md
Comment (thrbnhrtmnn) [inline TESTING.md:1]: We could also add a section about "Different Tests Types", explaining which types of tests (unit tests, integration tests, performance tests, e2e tests etc.) are part of the project.

Another section could explain our "Testing Strategy".

Both these sections would be more general and should be added before the Tooling section. A good example how this is done is the telemetry documentation, which you can see [here](https://github.com/deven-org/telemetry-functions/blob/develop/doc/TESTING.md).

**Review by thrbnhrtmnn — COMMENTED** [review]
Left a few comments. I think @ChristianHoffmannS2 should also do a review here :-)

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The project test suite is run with
+
+```sh
+$ yarn run test
+```
+
+## Contributing
+Contributions are welcome! If you find any issues or have suggestions for improvements, please open an issue or submit a
+pull request.
```
[the remark this answers is not in the document]
Comment (angsherpa456) [reply TESTING.md]: True, thanks and I liked it so adjusted it now in the testing file.

File: TESTING.md
[the remark this answers is not in the document]
Comment (angsherpa456) [reply TESTING.md:1]: Adjusted it now with the existing file.

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- [Testing](#testing)
+  - [Content](#content)
+  - [Tooling](#tooling)
+  - [open-wc/testing](#open-wc/testing)
+  - [How to write tests](#how-to-write-tests)
+  - [How to run tests](#how-to-run-tests)
+
+## Tooling
+
+## open-wc/testing
```
[the remark this answers is not in the document]
Comment (angsherpa456) [reply TESTING.md]: Adjusted now.

File: doc/TESTING.md [hunk 3]
```diff
@@ -1,48 +1,72 @@
...
-{ story: 'BlrTextButton--blr-text-button', parameters: ['icon:boilerChevronDownGreen'] }
+To write tests using `@open-wc/testing`, follow these steps:
+
+1. Import the necessary functions and utilities from `@open-wc/testing`.
+2. Write your test cases using the provided utilities such as `fixture`, `html`, `litFixture`, etc.
+
+## Directory Structure
+- `src/components/`: Contains the source code of the components and respective test for it.
+
+## Examples
```
Comment (thrbnhrtmnn) [inline doc/TESTING.md:37]: If "Directory Structure" and "Examples" these are sub-section of "How to write tests", you should make them a h3.

Also please add them to the Table of Content.

File: doc/TESTING.md [hunk 4]
```diff
@@ -1,48 +1,72 @@
...
-- [Tooling](#tooling)
-- [How to write tests](#how-to-write-tests)
-- [How to run tests](#how-to-run-tests)
+- [Testing](#testing)
+  - [Different kind of tests](#different-kind-of-tests)
+  - [Testing strategy](#testing-strategy)
+  - [Tooling](#tooling)
+     - [open-wc/testing](#open-wc/testing)
+  - [How to write tests](#how-to-write-tests)
+  - [How to run tests](#how-to-run-tests)
```
Comment (thrbnhrtmnn) [inline doc/TESTING.md:11]: Please also include "Content".

The link to open-wc/testing is not working. I the anchor would need to be "#open-wc-testing" for it to work.

Please also include "Contributing".

**Review by thrbnhrtmnn — CHANGES_REQUESTED** [review]

File: doc/TESTING.md (hunk 3 above)
Comment (angsherpa456) [reply doc/TESTING.md]: Adjusted now.

File: doc/TESTING.md (hunk 4 above)
Comment (angsherpa456) [reply doc/TESTING.md]: Adjusted now.

**Review by thrbnhrtmnn — APPROVED** [review]
Approved! Looks really nice now :-)

**Review by davidken91 — APPROVED** [review]

**Pull request merged**